clang/runtime/dpct-rt/include/dpct/dpl_extras/dpcpp_extensions.h

```diff
@@ -564,6 +564,47 @@ class radix_sort {
   uint8_t *_local_memory;
 };
 
+/// Load linear segment items into block format across threads
+/// Helper for Block Load
+template <int GROUP_THREADS, typename InputT, int ITEMS_PER_THREAD,
+          typename InputIteratorT>
+__dpct_inline__ void load_blocked(size_t linear_tid, InputIteratorT block_itr,
+                                  InputT (&items)[ITEMS_PER_THREAD]) {
+#pragma unroll
+  for (int ITEM = 0; ITEM < ITEMS_PER_THREAD; ITEM++) {
+    if ((linear_tid * ITEMS_PER_THREAD) + ITEM < GROUP_THREADS) {
+      items[ITEM] = block_itr[(linear_tid * ITEMS_PER_THREAD) + ITEM];
+    }
+  }
+}
+
+template <int GROUP_THREADS, typename InputT, int ITEMS_PER_THREAD,
+          typename InputIteratorT>
+__dpct_inline__ void load_striped(size_t linear_tid, InputIteratorT block_itr,
+                                  InputT (&items)[ITEMS_PER_THREAD]) {
+#pragma unroll
+  for (int ITEM = 0; ITEM < ITEMS_PER_THREAD; ITEM++) {
+    if (linear_tid + (ITEM * ITEMS_PER_THREAD) < GROUP_THREADS) {
+      items[ITEM] = block_itr[linear_tid + (ITEM * ITEMS_PER_THREAD)];
```

Comment:
Let me think on this, because item per thread is similar to the work items present in a work group. So loading in striped indicates work items should get the stride. Let me know if I am mistaken.

Reply:
I think ITEM * GROUP_THREADS is correct if we donot use the if condition. Since here I am using a range block load it implies the same thing . If we remove the if condition, then the correct way is ITEM * GROUP_THREADS

Reply:
Maybe I am misunderstanding the algorithm being implemented here but my understanding based on the name `load_striped` is that we load directly into items with a striped arrangement:

Say each thread is to load 4 elements (ITEMS_PER_THREAD=4) into it's local `items` array, the work-group size is 64, and `block_itr` consists of [0, 1, 2, 3, ... 255]. I would expect work-item 0 `items` to be `[0, 64, 128 192]`, work-item 1 to have `[1, 65, 129, 193]` and so on. This would require offsets of `linear_tid + (ITEM * GROUP_WORK_ITEMS)`.